sambamba/sort.d

```diff
@@ -121,11 +121,23 @@ class Sorter {
             _reads_capa = 1024;
             auto sz = BamRead.sizeof * _reads_capa;
             _reads = cast(BamRead*)std.c.stdlib.malloc(sz);
+            if (_reads is null) {
+                throw new Exception("alloc failed: no space for read pointers");
+            }
         }
 
         void clear() {
             _used = 0;
             _n_reads = 0;
+            if (_low_memory) {
+                auto realloc_storage = cast(ubyte*)std.c.stdlib.realloc(read_storage, max_sz / 2);
+                if (realloc_storage !is null) {
```

Comment:
Technically, this `realloc` can also fail (could reallocate via allocate/copy/free). I choose not to throw an exception here, but to keep printing errors. This is a situation in which we can proceed, but sub-optimally, and cannot automatically do anything to improve things. Spamming the user probably highlights the problem but results should still be correct.